packages/typegpu/src/data/vectorImpl.ts

```diff
@@ -0,0 +1,830 @@
+import type { SelfResolvable } from '../types';
+import type { VecKind } from './wgslTypes';
+
+export abstract class VecBase extends Array implements SelfResolvable {
+  abstract get kind(): VecKind;
```

Comment:
switched `kind` to a getter, so we don't add a string to each instance of the vector.